redisinsight/api/src/modules/triggered-functions/triggered-functions.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Command } from 'ioredis';
+import { HttpException, Injectable, Logger } from '@nestjs/common';
+import { catchAclError } from 'src/utils';
+import { concat } from 'lodash';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnectionService } from 'src/modules/database/database-connection.service';
+import { ShortLibrary, Library, Function } from 'src/modules/triggered-functions/models';
+import {
+  getLibraryInformation, getShortLibraryInformation, getLibraryFunctions,
+} from 'src/modules/triggered-functions/utils';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class TriggeredFunctionsService {
+  private logger = new Logger('TriggeredFunctionsService');
+
+  constructor(
+    private readonly databaseConnectionService: DatabaseConnectionService,
+  ) {}
+
+  /**
+   * Get library list for particular database with name, user, totalFunctions, pendingJobs fields only
+   * @param clientMetadata
+   */
+  public async libraryList(
+    clientMetadata: ClientMetadata,
+  ): Promise<ShortLibrary[]> {
+    let client;
+    try {
+      client = await this.databaseConnectionService.getOrCreateClient(clientMetadata);
+      const reply = await client.sendCommand(
+        new Command('TFUNCTION', ['LIST'], { replyEncoding: 'utf8' }),
+      );
+      const libraries = reply.map((lib: string[]) => getShortLibraryInformation(lib));
+      client.disconnect();
```

Comment:
please, remove all client.disconnect invocations because we will close client after each command execution.

Reply:
Sorry, my fault, forgot about it